tests/test_openeo_gfmap/test_unit_patch_extractors.py

```diff
@@ -0,0 +1,91 @@
+from openeo_gfmap.features import PatchFeatureExtractor
+from openeo.udf import XarrayDataCube
+
+import pytest
+import numpy as np
+import xarray as xr
+from unittest.mock import MagicMock, patch
```

Comment:
I think the openeo-python-client works with requests_mock. I haven't worked with unittest.mock yet but it might be nice to keep working with the same packages to ease development.
If this packafe has a clear advantage we might also consider using this package for the python client.

Reply:
I'll take a look. 

I thought requests_mock was restricted to https requests. In that sense we can mock loading data, but not sure if it enables one to mock intermediate results as well.

Reply:
I guess the usage of request_mock will be more in place for testing the s1 and S2 fetchers